types/wlr_linux_dmabuf.c

```diff
@@ -0,0 +1,431 @@
+#ifndef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 200809L
+#endif
+#include <assert.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <wayland-server.h>
+#include <wlr/render.h>
+#include <wlr/types/wlr_linux_dmabuf.h>
+#include <wlr/util/log.h>
+#include "linux-dmabuf-unstable-v1-protocol.h"
+
+static void wl_buffer_destroy(struct wl_client *client,
+	struct wl_resource *resource)
+{
+	wl_resource_destroy(resource);
+}
+
+static const struct wl_buffer_interface wl_buffer_impl = {
+	wl_buffer_destroy,
+};
+
+/* Return wlr_dmabuf_buffer if the given resource was created
+   via the linux-dmabuf protocol */
+struct wlr_dmabuf_buffer *wlr_linux_buffer_get(struct wl_resource *resource)
+{
+	struct wlr_dmabuf_buffer *buffer;
+
+	if (!resource) {
+		return NULL;
+	}
+
+	if (!wl_resource_instance_of(resource, &wl_buffer_interface,
+			&wl_buffer_impl)) {
+		return NULL;
+	}
+
+	buffer = wl_resource_get_user_data(resource);
```

Comment:
* We generally create `<type>_from_resource` helpers to provide better type safety (for instance, https://github.com/swaywm/wlroots/blob/master/types/wlr_gamma_control.c#L28). You'll be able to re-use this helper on line 64.
* Style nitpick: declare variables when they're first used